TestApp/Repositories/DataAccessRepositories.cs

```diff
@@ -0,0 +1,285 @@
+            }
+            catch (Exception ex)
+            {
+                return null;
+            }
+            
+        }
+        public IEnumerable<Person> GetPersonListByFirstOrLastName(string FName, string LName)
+        {
+            try
+            {
+                if (FName.Trim().ToString() == "" && FName.Trim().ToString() == "")
+                {
+                    var persons = ctx.Persons.Where(x=>x.isDeleted==false).ToList();
+                    return persons;
+                } else if(FName.Trim().ToString() == "")
+                {
+                    var persons = ctx.Persons.Where(x=>x.LastName==LName && x.isDeleted==false).ToList();
+                    return persons;
+                } else
+                {
+                    var persons = ctx.Persons.Where(x => x.FirstName== FName && x.isDeleted==false).ToList();
+                    return persons;
+                }
+
+            }
+            catch(Exception ex)
+            {
+                return null;
+            }
+        }
+        public IEnumerable<Person> GetPersonListBySpecificIdentity(string specification)
+        {
+            try
+            {
+                if (specification.Trim().ToString() == "")
+                {
+                    var persons = ctx.Persons.ToList();
+                    return persons;
+                }                
+                else
+                {
+                    var number = (int)((IdentificationType)Enum.Parse(typeof(IdentificationType), specification));
+                    var identities = ctx.Identifier.Distinct().Where(x => x.Value == number.ToString()).ToList();
+                    List<Person> personList = new List<Person>();
+                    if (identities != null)
+                    {
+                        foreach (var identity in identities)
+                        {
+                            Person person = new Person();
+                            person = ctx.Persons.Where(x => x.id == identity.Personid).FirstOrDefault();
+                            personList.Add(person);
+                        }
+                    }
+                                                            
+                    return personList;
+                }
+
+            }
+            catch (Exception ex)
+            {
+                return null;
+            }
+        }
+
+        public bool CreatePersonWithIdetifiers(Person person)
+        {
+            try
+            {
+                ctx.Persons.Add(person);
+                ctx.SaveChanges();
+                return true;
+                //    int res = ctx.SaveChanges();
+                //    return res;
+            }
+            catch (Exception ex)
+            {
+                return false;
+            }
+        }
+        public bool CreatePersonWithOutIdetifiers(Person person)
+        {
+            try
+            {
+                Person personObj = new Person();
```

Comment:
whats the difference between Person parameter and personObj?

Reply:
PersonObj is local object and Person object parameter object. we can not use only one Object when we need to update only specific .